packages/slate/src/commands/at-range.js

```diff
@@ -60,10 +60,12 @@ Commands.addMarkAtRange = (editor, range, mark) => {
       let index = 0
       let length = node.text.length
 
-      if (key === start.key) index = start.offset
       if (key === end.key) length = end.offset
-      if (key === start.key && key === end.key)
-        length = end.offset - start.offset
+
```

Comment:
is this change significant for the fix or is this just trying to tighten up logic?

Reply:
This logic is slightly different now specifically in the case where `(key === start.key && key !== end.key)`

Before:
`length = node.text.length`
Now:
`length = node.text.length - start.offset`

Reply:
Sure this is needed, if you move the offset you need to truncate the length accordingly - in both cases when key === and  !== end.key